### ISSUE TITLE
[DOC] compress_streams documentation is misleading

### ISSUE THREAD

**Issue by StefRe** [pikepdf/pikepdf#136]
In order to save a pdf file with uncompressed streams it's not sufficient to just specify `compress_streams=False`, as suggested by the documentation:
https://github.com/pikepdf/pikepdf/blob/7bbc01b4d523281c73407516769767cfa4ac2790/src/pikepdf/_methods.py#L649-L652

You also need to specify `stream_decode_level=pikepdf.StreamDecodeLevel.generalized`, otherwise (e.g. with the default `stream_decode_level=pikepdf.StreamDecodeLevel.none`) streams still get compressed even if `compress_streams=False`.

It took me a while to figure this out, eventually the [qpdf documentation](http://qpdf.sourceforge.net/files/qpdf-manual.html) put me on the right track:

>  uncompress: uncompress stream data compressed with generalized filters when possible; equivalent to --compress-streams=n --decode-level=generalized

I think this should be mentioned in the [documentation of `save`](https://pikepdf.readthedocs.io/en/latest/api/main.html?highlight=uncompress#pikepdf.Pdf.save) and the [docs of the `none` value for `pikepdf.StreamDecodeLevel`](https://pikepdf.readthedocs.io/en/latest/api/main.html?highlight=uncompress#pikepdf.StreamDecodeLevel.none) should be corrected.

Example to reproduce:
```
import pikepdf
import subprocess

in_file = 'dummy.pdf'  # from https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf
out_pike = 'dummy_uncompressed_pike.pdf'
out_qpdf = 'dummy_uncompressed_qpdf.pdf'

with pikepdf.open(in_file) as pdf:
    pdf.save(out_pike, stream_decode_level=pikepdf.StreamDecodeLevel.generalized, compress_streams=False)
    #pdf.save(out_pike, compress_streams=False) # --> this leaves streams compressed

subprocess.run(['qpdf.exe', '--stream-data=uncompress', in_file, out_qpdf])  # for comparison
```

If you like I could try and make a PR but English is not my native language, so maybe someone else can do it better.

**Comment by jbarlow83** [comment]
Yes, you are correct. What is actually happening is that any *new* streams you created and attached to the PDF will be saved uncompressed when `compress_streams=False`. To force decompression of existing streams, you do need to set the decode. I agree this is not clear in the documentation.

**Comment by jbarlow83** [comment]
Fixed in 1f103054
